ecc_enclave/enclave/shim.h

```diff
@@ -6,27 +6,130 @@
 
 #pragma once
 
+#include <stdbool.h>
 #include <map>
-#include <set>
 #include <string>
 #include <vector>
 
-typedef std::map<std::string, std::string> write_set_t;
-typedef std::set<std::string> read_set_t;
-typedef std::map<void*, std::pair<read_set_t*, write_set_t*>> context_t;
+typedef void* fpc_ctx_t;
 
-// shim put/get
-void get_state(const char* key, uint8_t* val, uint32_t max_val_len, uint32_t* val_len, void* ctx);
-void put_state(const char* key, uint8_t* val, uint32_t val_len, void* ctx);
+// TODO: Some Meta-questions
+// - should we prefix functions with fpc_ or alike to make them more distinct?
+//   Alternatively, we might go C++ and define a shim object like the go shim
```

Comment:
ack! this would be much closer to the existing go programming model; 
https://godoc.org/github.com/hyperledger/fabric/core/chaincode/shim#ChaincodeStubInterface

Reply:
That was also were i was coming for. But after discussion this morning i made executive decision that for now we leave out C++ and stay with C as (a) that's what we already have, (b) sooner or later (though probably rather later than sooner :-) we might go to a different language like WASM and (c) i'm not super-fond of (nor super-proficient in) in C++ ..